Source/Diagnostics/ComputeDiagFunctors/BackTransformParticleFunctor.cpp

```diff
@@ -0,0 +1,174 @@
+                              WarpXParticleContainer *pc_src,
+                              std::string species_name,
+                              int num_buffers)
+    : m_pc_src(pc_src), m_species_name(species_name), m_num_buffers(num_buffers)
+{
+    InitData();
+}
+
+
+void
+BackTransformParticleFunctor::operator () (ParticleContainer& pc_dst, int &totalParticleCounter, int i_buffer) const
+{
+    if (m_perform_backtransform[i_buffer] == 0) return;
+    auto &warpx = WarpX::GetInstance();
+    // get particle slice
+    const int nlevs = std::max(0, m_pc_src->finestLevel()+1);
+    auto tmp_particle_data = m_pc_src->getTmpParticleData();
+    int total_particles_added = 0;
+    for (int lev = 0; lev < nlevs; ++lev) {
+        amrex::Real t_boost = warpx.gett_new(0);
+        amrex::Real dt = warpx.getdt(0);
+
+        for (WarpXParIter pti(*m_pc_src, lev); pti.isValid(); ++pti) {
+            auto ptile_dst = pc_dst.DefineAndReturnParticleTile(lev, pti.index(), pti.LocalTileIndex() );
+        }
+
+        auto& particles = m_pc_src->GetParticles(lev);
+#ifdef AMREX_USE_OMP
+#pragma omp parallel
+#endif
+        {
+            // Temporary arrays to store copy_flag and copy_index for particles
+            // that cross the z-slice
+            amrex::Gpu::DeviceVector<int> FlagForPartCopy;
+            amrex::Gpu::DeviceVector<int> IndexForPartCopy;
+
+            for (WarpXParIter pti(*m_pc_src, lev); pti.isValid(); ++pti) {
+
+                auto index = std::make_pair(pti.index(), pti.LocalTileIndex());
+
+                const auto GetParticleFilter = SelectParticles(pti, tmp_particle_data,
+                                               m_current_z_boost[i_buffer],
+                                               m_old_z_boost[i_buffer]);
+                const auto GetParticleLorentzTransform = LorentzTransformParticles(
+                                                         pti, tmp_particle_data,
+                                                         t_boost, dt,
+                                                         m_t_lab[i_buffer]);
+
+                long const np = pti.numParticles();
+
+                FlagForPartCopy.resize(np);
+                IndexForPartCopy.resize(np);
+
+                int* const AMREX_RESTRICT Flag = FlagForPartCopy.dataPtr();
+                int* const AMREX_RESTRICT IndexLocation = IndexForPartCopy.dataPtr();
+
+                const auto& ptile_src = particles.at(index);
+                auto src_data = ptile_src.getConstParticleTileData();
+                // Flag particles that need to be copied if they cross the z-slice
+                // setting this to 1 for testing (temporarily)
+                amrex::ParallelFor(np,
+                [=] AMREX_GPU_DEVICE(int i)
+                {
+                    Flag[i] = GetParticleFilter(src_data, i);
+                });
+
+                const int total_partdiag_size = amrex::Scan::ExclusiveSum(np,Flag,IndexLocation);
```

Comment:
@atmyers this is the last line on Backtrace

Reply:
[Backtrace_Milestone.txt](https://github.com/ECP-WarpX/WarpX/files/7584931/Backtrace_Milestone.txt)
here is the Backtrace from the Debug=True simulation.